Add shortcut for "Lookup Symbol"

eterlan opened the pull request:
<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->
Hi! I would like to add a shortcut for lookup symbol, basically its all other's work.
I found someone do part of it a year ago #60156
Here is a pr related [#4577](https://github.com/godotengine/godot-proposals/issues/4577)
I test it and cannot find any problem.

Really appreciate you guys work. I'm new to Godot and first time make pull request, hopefully I'm doing this correctly, if anything wrong please just point it out, I would be more than happy to fix it.

Calinou committed:
Fix directional LightmapGI being too dark with static lights

The brightness now matches dynamic lights (indirect light baked only)
when Directional is enabled.

Co-authored-by: Priyansh Rathi <techiepriyansh@gmail.com>

Commit:
Add Duplicate Lines shortcut to CodeTextEditor

This keyboard shortcut has been made with inspiration from the VS Code keyboard shortcut editor.action.copyLinesDownAction. It duplicates all selected lines and inserts them below no matter where the caret is within the line.

bruvzg committed:
[RTL] Add support for image dynamic updating, padding, tooltips and size in percent.

MewPurPur committed:
Overhaul the Gradient Editor

DennisManaa committed:
made visual shader ports expandable by default if there is only one output port and it's of any vector type

Co-authored-by: QbieShay <cislaghi.ilaria@gmail.com>

raulsntos committed:
C#: Generate and use compat methods

- Implements `ClassDB::get_method_list_with_compatibility` to retrieve all methods from a class including compat methods.
- C# bindings generator now also generates compat methods.
- All generated C# methods now use `ClassDB::get_method_with_compatibility`.

398utubzyt committed:
C#: Abstract script class support

botero-dev committed:
Add transform support to CharFXTransform

Use absolute transforms for CharFX

fix formatting

DrRevert committed:
Allow to focus individual tabs in TabBar/TabContainer

Crimzoth committed:
Fix: Mass not set properly for RigidBody

Removed erroneous check, which caused _inv_mass not to be calculated when RigidBody2D or RigidBody3D used both custom center of mass and custom inertia.

EMBYRDEV committed:
Add layer, shadow and visibility range options to Scene Importer.

aaronfranke committed:
GLTF: Add GODOT_single_root extension

KoBeWi committed:
Free submenu children when clearing PopupMenu

Bromeon committed:
GDExtension: construct StringName directly from char*

So far, an indirection via String was necessary, causing at least 2 allocations and copies (String; String inside StringName).
Since StringNames often refer to string literals, this allows them to be directly constructed from C strings.

There are two formats: Latin-1 and UTF-8.

The Latin-1 constructor also provides the `p_is_static` flag: when the source has static storage duration, no copy/allocation will be needed.
However, the extension developer needs to uphold this lifetime guarantee.

raulsntos committed:
Expose the `TabBar` of a `TabContainer`

Add `TabContainer::get_tab_bar` method to retrieve the `TabBar` of a `TabContainer`.

dsnopek committed:
Implement reloading of GDExtensions

garychia committed:
Check if any global script class is shadowed by a variable

wscalf committed:
Allow readonly and writeonly C# properties to be accessed from GDScript

AThousandShips committed:
Disallow nested custom multiplayers in `SceneTree`

Enables clearing the custom multiplayer

AThousandShips committed:
Expose finding valid focus neighbors of a `Control` by side

Exposes the functionality used for ui navigation

smix8 committed:
Add 2D navigation mesh baking

Adds 2D navigation mesh baking.

BastiaanOlij committed:
Implement OpenXR Foveated rendering support

YuriSizov committed:
Differentiate between core and editor-only singletons

This makes sure that running scenes in debug mode
(from the editor) does not crash Godot.
In export mode it should already work correctly, because
editor-only singletons are never registered in the first place.

RandomShaper committed:
Optimize GDScript VM codegen for MSVC

akien-mga committed:
vulkan: Update all components to Vulkan SDK 1.3.261.1

Updates to volk, vulkan headers, `vk_enum_string_helper.h`, VMA,
glslang, spirv-reflect.

VMA doesn't tag SDK releases specifically, and still hasn't had a tagged
release since 3.0.1, but the Vulkan SDK now seems to ship a recent master
commit, so we do the same.

aitorciki committed:
[Windows] Fix not applying NVIDIA profile to new executables

An NVIDIA profile is applied to the current executable to disable
threaded OpenGL optimizations on Windows (see #71472). But because the
application is only added to the profile upon the profile creation,
newer executables won't be added to the profile (e.g. if the profile is
created on first launch of Godot_v4.1-stable_win64.exe, when users
update the editor and launch Godot_v4.2-stable_win64.exe, the profile
will never be applied to this new executable).
This patch fixes that scenario by splitting creating the profile (if it
doesn't exist) and adding the application (if it doesn't have a profile
applied) into two separate steps.
Applications that have been manually added to a different profile aren't
overriden to avoid confusing users who know what they're doing.

Sauermann committed:
Improve canvas layer index documentation

Document special case of identical canvas layer index.

Alex2782 committed:
Fixes the 'CLAMP' problem in the 'EditorSpinSlider' ...

... when arrow keys are pressed up or down.

bitsawer committed:
Fix shader language preprocessor include marker handling

DarioSamo committed:
Fix mipmap bias behavior by refactoring how samplers are created by Material Storage.

Introduces a new structure to store samplers created with certain parameters instead of storing a 'custom' set of samplers. Allows viewports to correctly configure the mipmap bias and use it when rendering the scene.

aaronfranke committed:
Add `is_conformal` method to Basis and Transform2D

Rubonnek committed:
Expose EditorInspector::get_edited_object to GDScript

jsjtxietian committed:
Ignore empty lines when commenting/uncommenting code

AThousandShips committed:
[Servers] Replace `ERR_FAIL_COND` with `ERR_FAIL_NULL` where applicable

AThousandShips committed:
[Modules] Replace `ERR_FAIL_COND` with `ERR_FAIL_NULL` where applicable

dsnopek committed:
Fix method hashes with default arguments

bitsawer committed:
Fix LightmapGI baking with GridMap

aaronfranke committed:
Doc: Reference String <-> PackedByteArray conversions from each other

dalexeev committed:
Editor: Remove unused Class Name field from Create Script dialog

timothyqiu committed:
Make editor support `--fullscreen` command-line argument

Ughuuu committed:
Update PinJoint2D API with angle limits and motor speed

add enabled methods for motor and angular limits
use correct name to get joint
update copyright

Calinou committed:
Clarify filtering by node type and group in the Scene tree dock

This functionality has been available since 4.0, but was difficult
to discover since there was no tooltip and the placeholder didn't
mention it.

bitsawer committed:
Fix BoneAttachment3D signal connection

akien-mga committed:
Web: Disable raycast module by default (no occlusion culling)

This means no CPU occlusion culling (and not compiling Embree), unless
you compile custom export templates with `module_raycast_enabled=yes`.

This reduces the memory footprint significantly, and binary size.

Fixes #70621.

Co-authored-by: Hugo Locurcio <hugo.locurcio@hugo.pro>

darksylinc committed:
Fix massive validation errors when enabling TAA + MSAA

TAA + MSAA would make Godot request unnecessary flags for an MSAA
velocity texture. flags that were not even actually needed.

This was causing:
 1. Unsupported GPUs to fail completely (e.g. Intel Arc 770)
 2. Wrong codepaths to be followed (causing validation errors, possibly
crashes or glitches)
 3. Unnecessary performance impact in all GPUs.

See
https://github.com/godotengine/godot/issues/71929#issuecomment-1722274359

Nomad1 committed:
Fixed VS 2022 Mac compatibility

Visual Studio 2022 on Mac marks the project as invalid if the project Guid is set. Easiest way to fix it is to remove outdated 'EnableGodotProjectTypeGuid' and other Guid references

0x4448 committed:
Omit quotes from completion if triggered with quote

Typing a single or double quote in an external editor triggers
auto-completion. The returned CompletionItem should not include
quotes since they're already in the editor.

CompletionParams was missing context in to_json() and this is
required to detect whether a quote was typed.

TokageItLab committed:
Improve retarget auto-mapping algorithm

groud committed:
Improve TileMap Y-sorting performance

Faless committed:
[Crypto] Fix generate_random_bytes for large chunks

Properly handle mbedtls errors, ensure we don't ask more bytes than
allowed for each iteration.

YuriSizov committed:
Replace flat buttons with flat-styled buttons with a visible pressed state

YuriSizov committed:
Allow to specify target folder when installing assets

This also changes the layout of the installer window to
better separate configuration of the installation and the
expected output.

marzecdawid committed:
Don't move not selected cursor item dragging files

CaTaTo23 committed:
Make all render driver project settings require restart

Update main/main.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

Update main/main.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

chokomancarr committed:
fix incorrect GL format code for 16 bit float formats

For FORMAT_XXXH half-precision format constants, the description uses GL_XXX32F, which is incorrect.
This fixes it to GL_XXX16F to align with the intended precision.

dalexeev committed:
GDScript: Prevent constructing and inheriting engine singletons

EIREXE committed:
Fix --gdscript-docs tool failing when autoloads are used in the project.

Fixes #79497

jsjtxietian committed:
Fix skeleton 3d editor's toolbar ui deleted from wrong container

dalexeev committed:
GDScript: Add `INFERRED_DECLARATION` warning

eterlan committed:
add shortcut for lookup symbol

eterlan committed:
add shortcut for lookup symbol

AThousandShips commented on editor/plugins/script_text_editor.cpp:
```diff
@@ -2419,6 +2420,8 @@ void ScriptTextEditor::register_editor() {
 	ED_SHORTCUT("script_text_editor/contextual_help", TTR("Contextual Help"), KeyModifierMask::ALT | Key::F1);
 	ED_SHORTCUT_OVERRIDE("script_text_editor/contextual_help", "macos", KeyModifierMask::ALT | KeyModifierMask::SHIFT | Key::SPACE);
 
+	ED_SHORTCUT("script_text_editor/symbol_lookup", TTR("symbol_lookup"),  Key::F12);
```
```suggestion
	ED_SHORTCUT("script_text_editor/symbol_lookup", TTR("symbol_lookup"), Key::F12);
```

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

akien-mga committed:
Merge pull request #61910 from Calinou/lightmapgi-static-directional-fix-brightness

Fix directional LightmapGI being too dark with static lights

akien-mga committed:
Merge pull request #81521 from dsnopek/method-bind-default-argument-hash-fix

Fix method hashes with default arguments

akien-mga committed:
Merge pull request #81716 from akien-mga/web-disable-raycast-embree

Web: Disable raycast module by default (no occlusion culling)

akien-mga committed:
Merge pull request #81350 from DarioSamo/mipmap-bias

Fix mipmap bias behavior by refactoring how samplers are created by Material Storage.

akien-mga committed:
Merge pull request #81802 from Nomad1/vs2022_mac_fix

Fixed VS 2022 Mac compatibility

akien-mga committed:
Merge pull request #82088 from DennisManaa/qbie/expand-ports

Visual Shaders: Make output-ports for vector types expandable by default

akien-mga committed:
Merge pull request #82098 from dalexeev/gds-prevent-engine-singleton-inheritance

GDScript: Prevent constructing and inheriting engine singletons

akien-mga committed:
Merge pull request #82116 from EIREXE/gdscript-docs-fix

Fix `--gdscript-docs` tool failing when autoloads are used in the project.

DarioSamo committed:
Add FidelityFX Super Resolution 2.2 (FSR 2.2.1) support.

Introduces support for FSR2 as a new upscaler option available from the project settings. Also introduces an specific render list for surfaces that require motion and the ability to derive motion vectors from depth buffer and camera motion.

Supreeeme committed:
Expose texture_create_from_extension to GDExtension

timothyqiu committed:
Fix leak when calling remove_control_from_menu_panel

Chaosus committed:
Fix CurveEdit crash when dragging the curve if it is null

AThousandShips commented on editor/plugins/script_text_editor.cpp:
```diff
@@ -2419,6 +2420,8 @@ void ScriptTextEditor::register_editor() {
 	ED_SHORTCUT("script_text_editor/contextual_help", TTR("Contextual Help"), KeyModifierMask::ALT | Key::F1);
 	ED_SHORTCUT_OVERRIDE("script_text_editor/contextual_help", "macos", KeyModifierMask::ALT | KeyModifierMask::SHIFT | Key::SPACE);
 
+	ED_SHORTCUT("script_text_editor/symbol_lookup", TTR("symbol_lookup"), Key::F12);
```
```suggestion
	ED_SHORTCUT("script_text_editor/symbol_lookup", TTR("Lookup Symbol"), Key::F12);
```

AThousandShips commented on editor/plugins/script_text_editor.cpp:
```diff
@@ -2239,6 +2239,7 @@ void ScriptTextEditor::_enable_code_editor() {
 	search_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/replace_in_files"), REPLACE_IN_FILES);
 	search_menu->get_popup()->add_separator();
 	search_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/contextual_help"), HELP_CONTEXTUAL);
+	search_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/symbol_lookup"), LOOKUP_SYMBOL);
```
```suggestion
	search_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/lookup_symbol"), LOOKUP_SYMBOL);
```

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

eterlan committed:
Update editor/plugins/script_text_editor.cpp

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

BlueCube3310 committed:
Disable bounding box shadows for advanced scene importer

dsnopek committed:
GDExtension: Remove redundant method bind hash check

AThousandShips committed:
Replace `radians` range hint with `radians_as_degrees`

KoBeWi committed:
Add call validation to CommandPalette

KoBeWi committed:
Make InputEventShortcut always pressed

YuriSizov committed:
Fix theme access and improve UX in AnimationTree editor

YuriSizov committed:
Check for type variations in inherited themes

deralmas committed:
Linux/OpenGL: don't force vsync in the editor

I couldn't tell whether this has an actual purpose and it feels more
like a debug remnant.

We also need to be able to disable vsync in the editor for the WIP
Wayland backend (in the EGL driver) as it does manual frame throttling.

BrunoArmondBraga committed:
Adds additional tests for RegEx class

Co-authored-by: Guilhermev25 <guilhermeferreira2529@hotmail.com>

Chubercik committed:
Remove the separator from ItemList's thumbnail mode

jaros3 committed:
[C#] Use `HashCode.Combine()` for basic composite types instead of xor

SaracenOne committed:
Make notify_dependency_error only defer calls if called from secondary threads.

akien-mga committed:
Sync controller mappings DB with SDL2 community repo

Synced with gabomdq/SDL_GameControllerDB@fc4b33c87239865487a3d956f7540a8d420066f5

Bromeon committed:
Fix platform-dependent hashing for char* and wchar_t*

Since char/wchar_t can be either signed or unsigned, its conversion to uint32_t leads to different values depending on platform.
In particular, the same string represented as char* (Latin-1; StringName direct construction) or uint32_t (UTF-8; constructed
via String) previously resulted in different hashes.

Bromeon committed:
More precise documentation for GDExtension String constructors (bytes vs. character count)

akien-mga committed:
Merge pull request #82245 from akien-mga/sync-gamecontrollerdb

Sync controller mappings DB with SDL2 community repo

akien-mga committed:
Merge pull request #71915 from MewPurPur/grid-ient

Overhaul the Gradient Editor

akien-mga committed:
Merge pull request #78580 from Bromeon/feature/gdextension-stringname-ctor

Add GDExtension function to construct StringName directly from char*

akien-mga committed:
Merge pull request #78803 from EMBYRDEV/additional-mesh-import-options

Add layer, shadow and visibility range options to the Scene importer

akien-mga committed:
Merge pull request #80227 from raulsntos/get_tab_bar

Expose the `TabBar` of a `TabContainer`

akien-mga committed:
Merge pull request #81251 from aitorciki/nvidia-profile-multiapp

[Windows] Fix not applying NVIDIA profile to new executables

akien-mga committed:
Merge pull request #81425 from Rubonnek/expose-inspector-methods-master

Expose `EditorInspector::get_edited_object` to GDScript

akien-mga committed:
Merge pull request #81675 from Calinou/editor-filter-nodes-type-clarify

Clarify filtering by node type and group in the Scene tree dock

akien-mga committed:
Merge pull request #81775 from darksylinc/matias-taa-msaa

Fix massive validation errors when enabling TAA + MSAA

akien-mga committed:
Merge pull request #82008 from KoTeYkA23/render-selector-restart-fix

Make all render driver project settings require restart

akien-mga committed:
Merge pull request #82050 from chokomancarr/patch-1

docs: Fix incorrect GL format code for 16 bit float formats

akien-mga committed:
Merge pull request #82131 from jsjtxietian/fix-skeleton-editor-deleted-in-wrong-container

Fix skeleton 3d editor's toolbar ui deleted from wrong container

akien-mga committed:
Merge pull request #82168 from Supreeeme/ext-fix

Expose `texture_create_from_extension` to GDExtension

akien-mga committed:
Merge pull request #82171 from timothyqiu/sep-leak

Fix leak when calling `remove_control_from_menu_panel`

akien-mga committed:
Merge pull request #82181 from Chaosus/curve_fix_crash

Fix CurveEdit crash when dragging the curve if it is null

akien-mga committed:
Merge pull request #82190 from BlueCube3310/aabb-import-shadows-disable

Disable bounding box shadows for advanced scene importer

akien-mga committed:
Merge pull request #82191 from dsnopek/gdextension-redundant-hash-check

GDExtension: Remove redundant method bind hash check

akien-mga committed:
Merge pull request #82203 from KoBeWi/shortcuts_under_pressure

Make InputEventShortcut always pressed

akien-mga committed:
Merge pull request #82210 from YuriSizov/editor-animation-uiux-cleanup

Fix theme access and improve UX in AnimationTree editor

akien-mga committed:
Merge pull request #82218 from YuriSizov/themes-vary-vary-duck-duck-goose

Check for type variations in inherited themes

akien-mga committed:
Merge pull request #82236 from Chubercik/item_list_separator_fix

Remove the separator from ItemList's thumbnails mode

darksylinc committed:
Fix gltf importer forcing vertex colors on all materials

The importer already checks if a mesh has vertex colors and enables
vertex colors on the material using it.

Before this fix, GLTF importer would force shader generation to use
vertex colors even if the scene did not have vertex colors at all, or
did not need them; causing inefficient shader and PSO generation.

aaronfranke committed:
Simplify Geometry3D tests

lyuma committed:
Avoid crash when generating LODs on meshes with non-finite vertices.

jsjtxietian committed:
Fix can't unset exported typed array element when the type is set to Node

akien-mga committed:
SCons: Fix Python 3.12 SyntaxError with regex escape sequences

dalexeev committed:
GDScript: Fix crash with `GDScriptNativeClass`

sadikkuzu committed:
Fix typo in `heuristic_euclidian` helper in AStarGrid2D

MewPurPur committed:
Add white Font and Mesh icons

akien-mga committed:
Windows: Remove `MSVC` define redundant with `_MSC_VER`

Cleanup unused MSVC code in FileAccessUnix.

akien-mga committed:
Merge pull request #76027 from AThousandShips/focus_direction

Expose finding valid focus neighbors of a `Control` by side

akien-mga committed:
Merge pull request #78573 from dalexeev/editor-create-script-class-name

Editor: Remove unused Class Name field from Create Script dialog

akien-mga committed:
Merge pull request #79104 from DrRevert/focusable_tabs_refactor

Allow to focus individual tabs in `TabBar`/`TabContainer`

akien-mga committed:
Merge pull request #79965 from KoBeWi/popup_menu_sidequest

Free submenu children when clearing PopupMenu

akien-mga committed:
Merge pull request #80881 from BastiaanOlij/openxr_foveation_ext

Implement OpenXR Foveated rendering support

akien-mga committed:
Merge pull request #81101 from 398utubzyt/dotnet/abstract-class-support

C#: Add abstract class support

akien-mga committed:
Merge pull request #81197 from DarioSamo/fsr2-rd

Add FidelityFX Super Resolution 2.2 (FSR 2.2.1) support.

akien-mga committed:
Merge pull request #81200 from RandomShaper/gd_vm_msvc

Optimize GDScript VM codegen for MSVC

akien-mga committed:
Merge pull request #81833 from 0x4448/fix-autocomplete-quotes

Fix autocomplete quotes

akien-mga committed:
Merge pull request #81939 from YuriSizov/gui-flat-and-depressed

Replace flat buttons with flat-styled buttons with a visible pressed state

akien-mga committed:
Merge pull request #82194 from KoBeWi/your_command_failed._Good_luck_finding_out_why

Add call validation to CommandPalette

akien-mga committed:
Merge pull request #82195 from AThousandShips/radian_fix

Replace `radians` range hint with `radians_as_degrees`

akien-mga committed:
Merge pull request #82285 from lyuma/nan_lods

Avoid crash when generating LODs on meshes with non-finite vertices.

akien-mga committed:
Merge pull request #82288 from aaronfranke/simplify-geometry3d-tests

Simplify Geometry3D tests

akien-mga committed:
Merge pull request #82290 from akien-mga/scons-python3.12-syntax-error-escape-sequences

SCons: Fix Python 3.12 SyntaxError with regex escape sequences

akien-mga committed:
Merge pull request #82297 from sadikkuzu/master

Fix typo in `heuristic_euclidian` helper in AStarGrid2D

akien-mga committed:
Merge pull request #80962 from YuriSizov/godot-the-editor-engine

Differentiate between core and editor-only singletons

akien-mga committed:
Merge pull request #82272 from darksylinc/matias-gltf-color

Fix gltf importer forcing vertex colors on all materials

akien-mga committed:
Merge pull request #82302 from MewPurPur/alt-font-icon

Add Font and Mesh icons that aren't grayed out

eterlan commented:
@AThousandShips  can it be merged? I use it a days, didn't notice any problem.

AThousandShips commented:
It will have to be reviewed by the relevant team, and tested, please have patience with the review process

YuriSizov commented on editor/plugins/script_text_editor.cpp:
```diff
@@ -2419,6 +2420,8 @@ void ScriptTextEditor::register_editor() {
 	ED_SHORTCUT("script_text_editor/contextual_help", TTR("Contextual Help"), KeyModifierMask::ALT | Key::F1);
 	ED_SHORTCUT_OVERRIDE("script_text_editor/contextual_help", "macos", KeyModifierMask::ALT | KeyModifierMask::SHIFT | Key::SPACE);
 
+	ED_SHORTCUT("script_text_editor/lookup_symbol", TTR("Lookup Symbol"), Key::F12);
```
This needs a sensible override for macOS, since F-buttons are not always easily accessible on that platform. Alternatively, we could add this shortcut without binding any key combination to it.

eterlan replied:
That's a valid point, however, IMO user can define the key they like，so this might not be very important I guess?
And also, my mac showing that visual studio also use F12 for both mac and windows. And mac user can always switch to fn-off-by-default mode.
So I think make it costomizable to keep persistentency with other shortcut might be better?

YuriSizov replied:
It's already customizable with your PR. However, we provide alternative bindings for macOS to avoid conflicts and work around macOS UX. Namely, F-keys are problematic for touchbar devices. You can see this is done in the same file and all around the codebase, and it needs to be done here as well.

MarioLiebisch committed:
Updated compiler version detection

This fixes multiple issues/inconsistencies around  `get_compiler_version()`:
* With no shell allocated, launching the compiler could fail even
  with proper paths being set.
* The return value was described as "an array of version numbers as ints",
  but the function actually returned a `Dictionary` (or `None`).
* Not all calls were properly handling a `None` return value in case of errors.
  On Windows this broke compiling for me since #81869 with default settings.
* Some calls defined inconsistent defaults/fallbacks (`0` or `-1`).

KoBeWi committed:
Fix ScriptCreateDialog not accepting on submit

Sauermann committed:
Fix type of notification_func

akien-mga committed:
Merge pull request #73813 from groud/improve_y_sort_performances

Greatly improve Y-sort performance on TileMaps

akien-mga committed:
Merge pull request #81884 from Faless/crypto/random_in_chunks

[Crypto] Fix `generate_random_bytes` for large chunks

akien-mga committed:
Merge pull request #82325 from MarioLiebisch/updated-compiler-version-detection

Updated compiler version detection

SaracenOne committed:
Add error checks for DirAccess creation.

zaevi committed:
Support dark mode on Android and iOS.

akien-mga committed:
Merge pull request #66553 from PucklaJ/duplicate_lines

Add Duplicate Lines shortcut to CodeTextEditor

akien-mga committed:
Merge pull request #77829 from AThousandShips/mp_fix

Disallow nested custom multiplayers in `SceneTree`

akien-mga committed:
Merge pull request #80284 from dsnopek/gdextension-hot-reload

Implement reloading of GDExtensions

akien-mga committed:
Merge pull request #80796 from smix8/navgenerator_2d_4.x

Add 2D navigation mesh baking

akien-mga committed:
Merge pull request #81219 from akien-mga/vulkan-sdk-1.3.261.1

vulkan: Update all components to Vulkan SDK 1.3.261.1

akien-mga committed:
Merge pull request #81270 from Sauermann/fix-canvas-layer-index-doc

Improve canvas layer index documentation

akien-mga committed:
Merge pull request #81278 from Alex2782/bugfix_EditorSpinSlider_clamp#81272

Fix clamping logic in `EditorSpinSlider`

akien-mga committed:
Merge pull request #81381 from bitsawer/fix_shader_include_markers

Fix shader language preprocessor include marker handling

akien-mga committed:
Merge pull request #81486 from jsjtxietian/Ignoring-empty-lines--when-uncommenting

Ignore empty lines when uncommenting code

akien-mga committed:
Merge pull request #81545 from bitsawer/fix_lightmap_gridmap_bake

Fix LightmapGI baking with GridMap

akien-mga committed:
Merge pull request #81608 from timothyqiu/fullscreen-editor

Make editor support `--fullscreen` command-line argument

akien-mga committed:
Merge pull request #81620 from YuriSizov/assets-remap-install-folder

Allow to specify target folder when installing assets

akien-mga committed:
Merge pull request #81695 from bitsawer/fix_bone_attachment_signal

Fix BoneAttachment3D signal connection

akien-mga committed:
Merge pull request #81843 from TokageItLab/improve-automapping

Improve retarget auto-mapping algorithm

akien-mga committed:
Merge pull request #82045 from marzecdawid/don't-move-not-selected-cursor-item-in-filesystem-dock

While dragging files don't move not selected cursor item in filesystem-dock

akien-mga committed:
Merge pull request #82225 from BrunoArmondBraga/improves_regextests

Adds additional tests for RegEx class

akien-mga committed:
Merge pull request #82230 from zaevi/dark-mode-android-ios

Support dark mode on Android and iOS.

akien-mga committed:
Merge pull request #82287 from jsjtxietian/fix-clear-error-when-unset-exported-typed-array-vai-editor

Fix can't unset exported typed array element when the type is set to Node

akien-mga committed:
Merge pull request #82294 from dalexeev/gds-fix-gdsnc-crash

GDScript: Fix crash with `GDScriptNativeClass`

akien-mga committed:
Merge pull request #82328 from KoBeWi/submit_to_the_dialog_overlords

Fix ScriptCreateDialog not accepting on submit

akien-mga committed:
Merge pull request #82332 from Sauermann/fix-struct-type

Fix type of `notification_func`

RPicster committed:
Optimizing glow behaviour

bruvzg committed:
Fix compiler detection.

jsjtxietian committed:
Fix can not set process priority of node if not any process is processing

bitsawer committed:
Add device info to GLES3 shader cache key hash

kleonc committed:
Fix animated tile time-slice calculation accumulating float errors

akien-mga committed:
Merge pull request #82352 from bruvzg/fix_compiler_det

Fix compiler detection.

akien-mga committed:
Merge pull request #77819 from 0xafbf/char-fx-tests

Make it possible to change character transform in RichTextEffect

akien-mga committed:
Merge pull request #78757 from Crimzoth/RigidBody3D-mass-issue-fix

Correctly set mass for a rigid body with custom inertia and center of mass

akien-mga committed:
Merge pull request #79523 from aaronfranke/is-conformal

Add `is_conformal` method to Basis and Transform2D

akien-mga committed:
Merge pull request #80410 from bruvzg/rtl_img_features

[RTL] Add support for image dynamic updating, padding, tooltips and size in percent.

akien-mga committed:
Merge pull request #80527 from raulsntos/dotnet/generate-compat-methods-from-classdb

C#: Generate and use compat methods

akien-mga committed:
Merge pull request #80587 from garychia/shadowed_class_name

Check if any global script class is shadowed by a variable

akien-mga committed:
Merge pull request #81564 from aaronfranke/doc-string-byte-array

Doc: Reference String <-> PackedByteArray conversions from each other

akien-mga committed:
Merge pull request #81610 from Ughuuu/add-angle-limits-and-motor-to-pin-joint-2d

Update PinJoint2D API with angle limits and motor speed

akien-mga committed:
Merge pull request #82221 from Riteo/linux-ogl-no-vsync-override

Linux/OpenGL: Don't force vsync in the editor

akien-mga committed:
Merge pull request #82240 from jaros3/vector2i-gethashcode

[C#] Use `HashCode.Combine()` for basic composite types instead of xor

akien-mga committed:
Merge pull request #82244 from SaracenOne/fix_dependency_error_popup

Fix missing dependency warning popup

akien-mga committed:
Merge pull request #82304 from akien-mga/windows-remove-redundant-MSVC-define

Windows: Remove `MSVC` define redundant with `_MSC_VER`

akien-mga committed:
Merge pull request #82313 from AThousandShips/null_check_servers

[Servers] Replace `ERR_FAIL_COND` with `ERR_FAIL_NULL` where applicable

akien-mga committed:
Merge pull request #82358 from jsjtxietian/fix-can-not-set-process-priority-of-node-in-editor

Fix can not set process priority of node in certain cases

kleonc committed:
Fix typo in dev assert in NavMeshGenerator2D

KoBeWi committed:
Fix submenus deleted accidentally

raulsntos committed:
C#: Remove compat method that is now generated

akien-mga committed:
Merge pull request #82368 from kleonc/NavMeshGenerator2D-dev-assert-assignment-typo

Fix typo in dev assert in NavMeshGenerator2D

akien-mga committed:
Merge pull request #82375 from raulsntos/dotnet/fix-compat

C#: Remove compat method that is now generated

akien-mga committed:
Merge pull request #67304 from wscalf/master

Allow readonly and writeonly C# properties to be accessed from GDScript

akien-mga committed:
Merge pull request #82139 from dalexeev/gds-add-inferred-declaration-warning

GDScript: Add `INFERRED_DECLARATION` warning

akien-mga committed:
Merge pull request #82347 from SaracenOne/dir_access_checks

Add error checks for DirAccess creation

akien-mga committed:
Merge pull request #82353 from RPicster/optimizing-glow

Optimizing glow behaviour

akien-mga committed:
Merge pull request #82359 from bitsawer/add_gles3_device_cache_key

Add device info to GLES3 shader cache key hash

YuriSizov committed:
Color match editor log toggles and flat pressed buttons

akien-mga committed:
Merge pull request #81760 from AThousandShips/null_check_modules

[Modules] Replace `ERR_FAIL_COND` with `ERR_FAIL_NULL` where applicable

akien-mga committed:
Merge pull request #81851 from aaronfranke/gltf-single-root

GLTF: Add root node export options and GODOT_single_root extension

akien-mga committed:
Merge pull request #82360 from kleonc/tilemap-tile-animation-time-float-error

Fix animated tile time-slice calculation accumulating float errors

akien-mga committed:
Merge pull request #82365 from YuriSizov/editor-colormatch-pressed-buttons

Color match editor log toggles and flat pressed buttons

akien-mga committed:
Merge pull request #82371 from KoBeWi/menus_that_should_be_there

Fix submenus deleted accidentally

eterlan committed:
add Mac shortcut override as CMD + M

eterlan committed:
Merge branch 'shortcut-for-lookup-symbol' of https://github.com/eterlan/godot into shortcut-for-lookup-symbol
